js/controllers/login.js

```diff
@@ -0,0 +1,111 @@
+/* global app */
+app.controller('loginCtrl', function($scope, $rootScope, $location, $state, enableEditService) {
+
+$scope.users=[
+    {
+        id: 1, 
+        username: "librarian", 
+        password: "librarian", 
+        name: "Librarian", 
+        active: true,
+        permission: 1
+    },
+    {
+        id: 2, 
+        username: "visitor", 
+        password: "visitor", 
+        name: "Wally West", 
+        active: true,
+        permission: 2
+    },
+    {
+        id: 3, 
+        username: "member", 
+        password: "member", 
+        name: "Clark Kent", 
+        active: true,
+        permission: 2
+    },
+    {
+        id: 4, 
+        username: "banned", 
+        password: "banned", 
+        name: "Evil Doer", 
+        active: false,
+        permission: 2
+    },
+    {
+        id: 5, 
+        username: "admin", 
+        password: "admin", 
+        name: "administrator", 
+        active: true,
+        permission: 1
+    }
+];
+
+$scope.logInMessage = "";
+$scope.errorMessage = "";
+$rootScope.logInId = "";
+$rootScope.logInPermission = "";
+$rootScope.logInActive = "";
+
+$scope.loginMatch = function() {
```

Comment:
this function looks good - clear logic and seems like it should do what it is supposed to in an efficient way